client/src/client-only-routes/show-project-links.tsx

```diff
@@ -152,6 +147,14 @@ const ShowProjectLinks = (props: ShowProjectLinksProps): JSX.Element => {
       }
     : null;
 
+  const isCertName = (
+    maybeCertName: string
+  ): maybeCertName is keyof typeof fullProjectMap | 'Legacy Full Stack' => {
+    if (maybeCertName === 'Legacy Full Stack') return true;
+    return maybeCertName in fullProjectMap;
+  };
+  if (!isCertName(certName)) return <div> Unknown Certification</div>;
```

Comment:
As far as I can see, this is unreachable?

If a cert is invalid, the api responds with a flash: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/api-server/src/server/boot/certificate.js#L393

If the server responds with a flash, the client redirects: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/redux/show-cert-saga.js

---

Only reason I bring this up, is we should have this typed and tested in such a way we do not need this check, because we never want Campers to see such a message.

Reply:
I agree. The only issue is I want to a) convince TS  of the types (safely) and b) cover the case where the input is somehow wrong.

Ideally we want all type-checking/validation to be on the boundaries (like we do in the new API), but this is the best I've come up with so far.